source/future.c

```diff
@@ -47,12 +47,11 @@ struct aws_future_impl {
         aws_future_impl_result_release_fn *release;
     } result_dtor;
     int error_code;
-    /* sum of bit fields should be 32 */
-#define BIT_COUNT_FOR_SIZEOF_RESULT 27
```

Comment:
update the code down at line 111:
```
/* we store sizeof_result in a bit field, ensure the number will fit */
    AWS_ASSERT(sizeof_result <= (UINT_MAX >> (32 - BIT_COUNT_FOR_SIZEOF_RESULT)));
future->sizeof_result = (unsigned int)sizeof_result;
```

debatable whether we change this to `size_t` just for simplicity, or keep it smaller just for smaller's sake 🤷‍♀️. your call.